src/compiler/checker.ts

```diff
@@ -17897,6 +17899,9 @@ namespace ts {
                     errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_does_not_exist_on_type_1_Did_you_mean_2, declarationNameToString(propNode), typeToString(containingType), suggestedName);
                     relatedInfo = suggestion.valueDeclaration && createDiagnosticForNode(suggestion.valueDeclaration, Diagnostics._0_is_declared_here, suggestedName);
                 }
+                else if (isStaticPropertyExists(propNode, containingType)) {

```

Comment:
I think finding a static property is probably better than finding a possible typo, and about the same as finding it on the promised type. So probably this should go before the spelling suggestion check and possibly before the promised type check (if it's more convenient there).